### PR TITLE
fix thumbnail storage location

### LINKED ISSUE #5913: Remote dynamic thumbnails are stored at the wrong location
### Description

The remote thumbnails path and filename are derived from `media_id` when they're generated and stored with `dynamic_thumbnails` set to `true`. The said thumbnails are looked up by `filesystem_id` which fails every time. Synapse uses a lot of CPU regenerating the thumbnails all the time. When `dynamic_thumbnails` option is then turned off Synapse can't find the thumbnails generated when the said option was turned on and just returns 404.

### Steps to reproduce

- set `dynamic_thumbnails` to `true`
- try to load a thumbnail of a remote media
- now set `dynamic_thumbnails` to `false`
- try to load the same thumbnail of the remote media
- it returns 404

Local media thumbnails are generated and retrieved just fine. For example, I'm trying to retrieve a media with id `zOUYnsGQKLQypwBoQkRpnkXh` and corresponding filesystem id `WFzEWtlvXlCaZShcaIthgzrG`. The media itself is stored at `/var/lib/matrix-synapse/media/bc/remote_content/matrix.org/WF/zE/WtlvXlCaZShcaIthgzrG` and I can retrieve it from my server. The thumbnail should be located at `/var/lib/matrix-synapse/media/bc/remote_thumbnail/matrix.org/WF/zE/WtlvXlCaZShcaIthgzrG/30-30-image-png` but it's actually stored at `/var/lib/matrix-synapse/media/bc/remote_thumbnail/matrix.org/zO/UY/nsGQKLQypwBoQkRpnkXh/30-30-image-png`. Why do we even need two identifiers? It confuses things pretty much (it's hard to find an offending file in the storage if you know its id without looking it up in the database) and doesn't add any security.

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: own server

<!--
What version of Synapse is running?
You can find the Synapse version by inspecting the server headers (replace matrix.org with
your own homeserver domain):
$ curl -v https://matrix.org/_matrix/client/versions 2>&1 | grep "Server:"
-->
- **Version**: 1.3.1+buster1

- **Install method**: apt repository
<!-- examples: package manager/git clone/pip  -->

- **Platform**: Debian stable amd64
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

### PR DESCRIPTION
**Pull request by L0ric0**
fixes #5913

Signed-off-by: Lorenz Steinert <lorenz@steinerts.de>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

**Comment by richvdh** [comment]
@L0ric0 please could you merge latest `develop` into your branch, to help the CI?

File: changelog.d/5915.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+The right filesystempath is now used to store thumbnails. Fix reported by rkfg.
```
Comment (richvdh) [inline changelog.d/5915.bugfix]: please could you phrase this in terms of the change in behaviour that the user will see?

(as per my questions on #5913, I remain confused as to what bug we are trying to fix here)

Comment (L0ric0) [reply changelog.d/5915.bugfix]: have changed it. Can you take a look if that is more fitting?

Comment (richvdh) [reply changelog.d/5915.bugfix]: sounds about right to me. thanks!

**Comment by richvdh** [comment]
I think the CI failures are due to an out-of-date branch; merging anyway.

**Pull request merged**